src/core/segment_buffers/inventory/segment_inventory.ts

```diff
@@ -318,7 +339,11 @@ export default class SegmentInventory {
                 bufferType, inventoryIndex, inventory.length);
       const removed = inventory.splice(inventoryIndex, inventory.length - inventoryIndex);
       for (const seg of removed) {
-        if (seg.bufferedStart === undefined && seg.bufferedEnd === undefined) {
+        if (
+          seg.bufferedStart === undefined &&
+          seg.bufferedEnd === undefined &&
+          seg.status === ChunkStatus.Failed
```

Comment:
> This PR fixes the issue by clearly identifying in the SegmentInventory chunks whose push operation had an issue. Those are:
> 
> never added to the BufferedHistory

So segments with status `Failed` here are added to the buffer history.
Did you made a typo and wanted to write `seg.status !== ChunkStatus.Failed` ?

Reply:
:O You're right, I ended up doing the opposite of what I wanted, I should probably add tests here